plutarch-test/src/Util.hs

```diff
@@ -0,0 +1,286 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Util (
```

Comment:
You want to move this to `Plutarch.Test.Property` inasmuch as the whole thing is to do with property testing.